src/LOAuth.ts

```diff
@@ -89,15 +96,28 @@ class LOAuth {
       }
 
       // If after login flow, decode from state
-      const decodedState = decodeURIComponent(queryParameters.get('state') || '{}');
+      const decodedState = decodeURIComponent(
```

Comment:
I know it's not new, but this line feels like it might be contributing to some of the problems with double encoding. Here is a test I did in a Node REPL:
```
new url.URLSearchParams('?test=%40').get('test')
```
The result of that was '@' implying that the `URLSearchParams` utility already handles URL decoding